src/fu-engine.c

```diff
@@ -8243,12 +8243,15 @@ fu_engine_load(FuEngine *self, FuEngineLoadFlags flags, FuProgress *progress, GE
 
 #ifdef HAVE_PASSIM
 	/* connect to passimd */
-	if (!passim_client_load(self->passim_client, &error_passim))
-		g_debug("failed to load Passim: %s", error_passim->message);
-	if (passim_client_get_version(self->passim_client) != NULL) {
-		fu_engine_add_runtime_version(self,
-					      "org.freedesktop.Passim",
-					      passim_client_get_version(self->passim_client));
+	if (!(fu_engine_config_get_p2p_policy(self->config) & FU_P2P_POLICY_NOTHING)) {
```

Comment:
I'm not so sure about this; we need to ensure the passim client is loaded (and unloaded?) if the config file changes at runtime. Give me a few hours and I can ponder.

Reply:
Alternate proposal in https://github.com/fwupd/fwupd/pull/8286 ?